src/io/perun/markdown.clj

```diff
@@ -85,20 +83,15 @@
          (.markdownToHtml processor))))
 
 (defn process-file [file options]
-  (let [file-content (slurp file)]
-    ; .getName returns only the filename so this should work cross platform
-    (u/info "Processing Markdown: %s\n" (.getName file))
-    (merge (parse-file-metadata file-content)
-           {:content (markdown-to-html file-content options)
-            ; todo: we will need to remove this from task
-            ; it's duplication of what base task is doing now
-            :path (:path file)
-            :filename (.getName file)})))
+  (let [iofile (io/file (:full-path file))
+        file-content (slurp iofile)
+        md-metadata (parse-file-metadata file-content)
+        html (markdown-to-html file-content options)]
+        ; .getName returns only the filename so this should work cross platform
+        (u/info "Processing Markdown: %s\n" (:filename file))
+        (merge md-metadata {:content html} file)))
```

Comment:
I think it's best to print messages like "Processing Markdown..." **before** the actual processing so if it fails it's clear which file caused the failure.